src/Knp/Component/Pager/Event/Subscriber/Sortable/ArraySubscriber.php

```diff
@@ -2,82 +2,94 @@
     {
-        if (is_array($event->target) && count($event->target) > 1)
-        {
-            if (isset($_GET[$event->options['sortFieldParameterName']])) {
-                $this->sortDirection = isset($_GET[$event->options['sortDirectionParameterName']]) && strtolower($_GET[$event->options['sortDirectionParameterName']]) === 'asc' ? 'asc' : 'desc';
-
-                // TODO add whitelist
-//                if (isset($event->options['sortFieldWhitelist'])) {
-//                    if (!in_array($_GET[$event->options['sortFieldParameterName']], $event->options['sortFieldWhitelist'])) {
-//                        throw new \UnexpectedValueException("Cannot sort by: [{$_GET[$event->options['sortFieldParameterName']]}] this field is not in whitelist");
-//                    }
-//                }
-
-                $sortFieldParameterName = explode('.', $_GET[$event->options['sortFieldParameterName']]);
-                if(isset($sortFieldParameterName[1])) {
-                    // Capitalize first letter in order to prepare getter construction
-                    $sortFieldName = ucfirst($sortFieldParameterName[1]);
-
-                    $this->currentSortingFieldGetter = "get{$sortFieldName}";
-
-                    // Getter detection
-                    $class = new ReflectionClass(get_class(current($event->target)));
-                    if($class->hasMethod($this->currentSortingFieldGetter)) {
-                        // Sort
-                        usort($event->target, array($this, "sort" . ucfirst($this->sortDirection)));
-                    }
-                }
-            }
+        if (!$accessor && class_exists('Symfony\Component\PropertyAccess\PropertyAccess')) {
+            $accessor = PropertyAccess::createPropertyAccessorBuilder()->enableMagicCall()->getPropertyAccessor();
         }
+
+        $this->propertyAccessor = $accessor;
     }
 
-    /**
-     * @param $object1 first object to compare
-     * @param $object2 second object to compare
-     */
-    private function sortAsc($object1, $object2)
+    public function items(ItemsEvent $event)
     {
-        $fieldValue1 = strtolower($object1->{$this->currentSortingFieldGetter}());
-        $fieldValue2 = strtolower($object2->{$this->currentSortingFieldGetter}());
-        if ($fieldValue1 == $fieldValue2) {
-            return 0;
+        if (!is_array($event->target) || empty($_GET[$event->options['sortFieldParameterName']])) {
+            return;
+        }
+
+        if (isset($event->options['sortFieldWhitelist']) && !in_array($_GET[$event->options['sortFieldParameterName']], $event->options['sortFieldWhitelist'])) {
+            throw new \UnexpectedValueException("Cannot sort by: [{$_GET[$event->options['sortFieldParameterName']]}] this field is not in whitelist");
+        }
+
+        if (isset($_GET[$event->options['sortDirectionParameterName']])) {
+            $this->sortDirection = strtolower($_GET[$event->options['sortDirectionParameterName']]);
```

Comment:
I've changed it because end result is same. Sort function works with "asc" only, anything else (even null) is sorted as if it was "desc". But since you prefer to cast it to "desc" before, I'll change it back.